lib/pengine/unpack.c

```diff
@@ -88,11 +88,11 @@ is_dangling_guest_node(pcmk_node_t *node)
     /* we are looking for a remote-node that was supposed to be mapped to a
      * container resource, but all traces of that container have disappeared 
      * from both the config and the status section. */
-    if (pe__is_guest_or_remote_node(node) &&
-        node->details->remote_rsc &&
-        node->details->remote_rsc->container == NULL &&
-        pcmk_is_set(node->details->remote_rsc->flags,
-                    pcmk_rsc_removed_filler)) {
+    if (pcmk__is_pacemaker_remote_node(node)
```

Comment:
alternatively
```
     if (pcmk__is_remote_node(node)
         && (node->details->remote_rsc != NULL)
         && pcmk_is_set(node->details->remote_rsc->flags,
                        pcmk_rsc_removed_filler)) {
```
We check `remote_rsc` twice that way, but it could be perceived as clearer in addition to saving a line

Reply:
Actually that would be misleading. We're looking for a guest node, which `pcmk__is_remote_node()` is intentionally designed to exclude (even though it lines up for this case).

Reply:
Ah, I was making an assumption based on the beginning of the comment ("we are looking for a remote-node...").

The rest of the comment makes it clearer that we're looking for a guest node, but it could probably be more explicit.

Reply:
Yep, I'll leave that until this is moved into libcrmcommon